src/genai/eyfs/run_classifier.py

```diff
@@ -0,0 +1,113 @@
+"""Classify the BBC activities to EYFS areas of learning."""
+
+import asyncio
+import os
+import time
+
+import openai
+import pandas as pd
+
+from aiohttp import ClientSession
+from dotenv import load_dotenv
+
+from genai import FunctionTemplate
+from genai import MessageTemplate
+from genai.eyfs import EYFSClassifier
+from genai.utils import batch
+from genai.utils import read_json
+
+
+load_dotenv()
+
+# You need to create this manually before running the script
+# TODO: Create the subdir if not exists
+OUTPUT_FILENAME = "data/eyfs_labels/"
+PATH_TO_BBC_ACTIVITIES = "data/eyfs/tiny_happy_people - final - tiny_happy_people - final.csv"
+PATH_TO_AREAS_OF_LEARNING = "src/genai/eyfs/areas_of_learning.json"
+PATH_TO_MESSAGE_PROMPT = "src/genai/eyfs/prompts/classifier.json"
+PATH_TO_FUNCTION = "src/genai/eyfs/prompts/classifier_function.json"
+openai.api_key = os.environ["OPENAI_API_KEY"]
+
+
+def get_bbc_activities(path: str) -> pd.DataFrame:
+    """Read and clean the BBC activities file and return a dataframe."""
+    df = pd.read_csv(path)
+    df = df.rename(columns={"Age Range (if applicable)": "Age"})
+    df = df.dropna(subset=["text", "URL"])
+    df = df.drop_duplicates(subset=["URL"])
+
+    return df
+
+
+def get_areas_of_learning(path: str) -> tuple:
+    """Get the EYFS areas of learning and their text."""
+
+    # keys for the classes, text for the prompt
+    areas_of_learning = read_json(path)
+    areas_of_learning_text = "\n".join([v for v in areas_of_learning.values()])
+    areas_of_learning_keys = list(areas_of_learning.keys())
+
+    # Add "None" to cover cases where no option is applicable.
+    areas_of_learning_keys.append("None")
+
+    return areas_of_learning_keys, areas_of_learning_text
+
+
+async def main() -> None:
+    """Create prompts for path selection and infer paths."""
+    openai.aiosession.set(ClientSession())
+
+    # Fetch the BBC activities
+    df = get_bbc_activities(PATH_TO_BBC_ACTIVITIES)
+
+    # Fetch the EYFS areas of learning
+    _, areas_of_learning_text = get_areas_of_learning(PATH_TO_AREAS_OF_LEARNING)
+
+    print(f"Number of BBC activities: {len(df)}")  # noqa: T001
+
+    message = MessageTemplate.load(PATH_TO_MESSAGE_PROMPT)
+    function = FunctionTemplate.load(PATH_TO_FUNCTION)
+    model = "gpt-3.5-turbo"
+    temperature = 0.6
+
+    for i, batched_results in enumerate(batch(df, 20)):
```

Comment:
From `utils` it looks like batch works with lists, but here you're using it with dataframes - I guess it'll be ok as long as the dataframe has a "normal" index, ie 0, 1, 2, ... ?